parquet-column/src/main/java/org/apache/parquet/filter2/predicate/SchemaCompatibilityValidator.java

```diff
@@ -170,6 +174,24 @@ public Void visit(Not not) {
 
   private <T extends Comparable<T>> void validateColumn(Column<T> column) {
     ColumnPath path = column.getColumnPath();
+    HashSet<Type.ID> ids = new HashSet<>();
```

Comment:
nit: this can be introduced after the `if`

Reply:
Fixed. Thanks!